portality/bll/services/query.py

```diff
@@ -117,6 +126,15 @@ def search(self, domain, index_type, raw_query, account, additional_parameters):
 
         return res
 
+    def prepare_query(self, domain, index_type, raw_query, account):
```

Comment:
After a bit of reflection, I'm against this approach.  The reason being that it violates the encapsulation of the queryService.  The whole point of the query service is that it does all the work for you, and if we prepare a query and then execute it outside that service, then we risk duplicating code and introducing unexpected bugs.  

In this case, for example, this means that something that uses the `prepare_query` method misses applying the post-search filters that would have been applied if the main `search` method was used.  Right now that wouldn't matter as there aren't any post search filters, but were we to add any in the settings, they would not be applied like they would with all the other query endpoints.

Instead what we should do here is modify the `search` query so that `additional_parameters=None` in the signature, then just use `queryService.search` directly in the API code - I made a note above somewhere where I'd do that.

Reply:
Function removed